frontend/src/lib/components/input/Input.type.ts

```diff
@@ -0,0 +1,109 @@
+import type {SvelteHTMLElements} from 'svelte/elements';
+import type {Photo} from '$lib/types/candidateAttributes';
+
+export type InputProps =
+  | ({
+      type: 'text';
+    } & InputPropsBase<string> &
+      SvelteHTMLElements['input'])
+  | ({
+      type: 'text-multilingual';
+    } & InputPropsBase<LocalizedString> &
+      SvelteHTMLElements['input'])
+  | ({
+      type: 'textarea';
+    } & InputPropsBase<string> &
+      SvelteHTMLElements['textarea'])
+  | ({
+      type: 'textarea-multilingual';
+    } & InputPropsBase<LocalizedString> &
+      SvelteHTMLElements['textarea'])
+  | ({
+      type: 'number';
+    } & InputPropsBase<number> &
+      SvelteHTMLElements['input'])
+  | ({
+      type: 'date';
+    } & InputPropsBase<Date> &
+      SvelteHTMLElements['input'])
+  | ({
+      type: 'boolean';
+    } & InputPropsBase<boolean> &
+      SvelteHTMLElements['input'])
+  | ({
+      type: 'image';
+    } & InputPropsBase<Photo> &
+      SvelteHTMLElements['input'])
+  // TODO: Possibly implement
+  // | ({
+  //   type: 'image-multiple';
+  //   isOrdered?: boolean;
+  // } & InputPropsBase<Array<Photo>> & SvelteHTMLElements['input'])
+  | ({
+      type: 'select';
+      options: Array<Option>;
+    } & InputPropsBase<Key> &
+      SvelteHTMLElements['select'])
+  | ({
+      type: 'select-multiple';
+      options: Array<Option>;
+      isOrdered?: boolean;
+    } & InputPropsBase<Array<Key>> &
+      SvelteHTMLElements['select']);
+
+export type Key = string;
+
+type Option = {
+  key: Key;
+  label: string;
+};
+
+type InputPropsBase<TValue> = {
+  /**
+   * The label to show for the input or group of inputs if `multilingual`.
+   */
+  label: string;
+  /**
+   * Any additional props to be passed to the container element of the input. @default {}
+   */
+  containerProps?: SvelteHTMLElements['div'];
+  /**
+   * The id of the input. If not provided, a unique id will be generated.
+   */
+  id?: string;
+  /**
+   * Additional info displayed below the input.
+   */
+  info?: string;
+  /**
+   * If `locked` the input will be disabled and a lock icon is displayed.
+   */
+  locked?: boolean;
```

Comment:
Instead of having `locked` and `disabled` separately maybe there could be something like `state: 'enabled' | 'disabled' | 'locked'` because I don't know if it makes sense that e.g. `disabled` can be changed while `locked = true`.

Reply:
True, it's a bit superfluous, but it was included due to `disabled` being available by default as part of `restProps` forwarding to `input` elements. There might be a use case, though, for the field being temporarily `disabled` and permanently `locked` with the latter being accompanied with the icon as well. Otherwise, we could lose the whole `locked` prop and just use `disabled`.